tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java

```diff
@@ -52,4 +63,37 @@ protected void assertAdditionalMessageProperties(final Message msg) {
 
         assertThat(msg.isDurable()).isTrue();
     }
+
+    /**
+     * Checks an event with an unsupported device QoS level is rejected.
+     *
+     * @param ctx The test context.
+     *
+     * @throws InterruptedException if the test fails.
+     */
+    @Test
+    public void testEventIsRejectedForUnsupportedQosLevel(final VertxTestContext ctx) throws InterruptedException {
+        final VertxTestContext setup = new VertxTestContext();
+        final Tenant tenant = new Tenant();
+        final MultiMap requestHeaders = MultiMap.caseInsensitiveMultiMap()
+                .add(HttpHeaders.CONTENT_TYPE, "text/plain")
+                .add(HttpHeaders.AUTHORIZATION, authorization)
+                .add(HttpHeaders.ORIGIN, ORIGIN_URI)
+                .add(Constants.HEADER_QOS_LEVEL, String.valueOf(QoS.AT_MOST_ONCE.ordinal()));
+
+        helper.registry.addDeviceForTenant(tenantId, tenant, deviceId, PWD).onComplete(setup.completing());
+
+        assertThat(setup.awaitCompletion(5, TimeUnit.SECONDS)).isTrue();
+        if (setup.failed()) {
+            ctx.failNow(setup.causeOfFailure());
+            return;
+        }
+
+        httpClient.create(
+                getEndpointUri(),
+                Buffer.buffer("hello"),
+                requestHeaders,
+                ResponsePredicate.status(HttpURLConnection.HTTP_BAD_REQUEST))
```

Comment:
Note that there was a code change which leads to a different behaviour in Hono's AMQP adapter in 1.5.:
Previously the message was accepted (and the QoS was ignored), now it is rejected.

It's not specified as far as I can see how the adapter will react on this, but this may impact client if they had accidentally set a _qos_level_ HTTP header for events to AT_MOST_ONCE.

Reply:
you mean HTTP adapter, right?

Reply:
Yes, of course 😅 Sorry for the confusion.